ianalyzer_readers/readers/rdf.py

```diff
@@ -64,5 +78,19 @@ def document_subjects(self, graph: Graph) -> Iterable[Union[BNode, Literal, URIR
         '''
         return graph.subjects()
 
-    def _document_from_subject(self, graph: Graph, subject: Union[BNode, Literal, URIRef]) -> dict:
-        return {field.name: field.extractor.apply(graph=graph, subject=subject) for field in self.fields}
+    def _document_from_subject(self, graph: Graph, subject: Union[BNode, Literal, URIRef], metadata: dict) -> dict:
+        return {field.name: field.extractor.apply(graph=graph, subject=subject, metadata=metadata) for field in self.fields}
+
+
+def get_uri_value(node: URIRef) -> str:
+    ''' a utility function to extract the last part of a uri
+    For instance, if the input is URIRef('https://purl.org/mynamespace/ernie'),
+    the function will return 'ernie'
+
+    Parameters:
+        node: an URIRef input node
+    
+    Returns:
+        a string with the last element of the uri
+       '''
+    return node.n3().strip('<>').split('/')[-1]
```

Comment:
So far, all functions I've seen from `urllib.parse` seem surprisingly unhelpful (I've investigated this as a way to contruct urls in a similar fashion as `os.path.join` at some point). It can deal with `https://purl.org/mynamespace#ernie` but not with `https://purl.org/mynamespace/ernie`. It also wouldn't split off `<>` parts of a string.